src/sentry/api/endpoints/organization_member_team_details.py

```diff
@@ -150,6 +187,58 @@ def post(
 
         return Response(serialize(team, request.user, TeamWithProjectsSerializer()), status=201)
 
+    def put(
+        self,
+        request: Request,
+        organization: Organization,
+        member: OrganizationMember,
+        team_slug: str,
+    ) -> Response:
+        try:
+            team = Team.objects.get(organization=organization, slug=team_slug)
+        except Team.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        try:
+            omt = OrganizationMemberTeam.objects.get(team=team, organizationmember=member)
+        except OrganizationMemberTeam.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        serializer = OrganizationMemberTeamSerializer(data=request.data, partial=True)
+        if not serializer.is_valid():
+            return Response(status=400)
+        result = serializer.validated_data
+
+        if "role" in result and features.has("organizations:team-roles", organization):
+            new_role_id = result["role"]
+            try:
+                new_role = team_roles.get(new_role_id)
+            except KeyError:
+                return Response(status=400)
+
+            if not self._can_set_team_role(request, team, new_role):
+                return Response({"detail": ERR_INSUFFICIENT_ROLE}, status=400)
+
+            self._change_team_member_role(omt, new_role)
+
+        return Response(serialize(team, request.user, TeamWithProjectsSerializer()), status=200)
+
+    @staticmethod
+    def _change_team_member_role(member: OrganizationMemberTeam, role: TeamRole) -> None:
```

Comment:
nit: could we name this parameter `team_role`?

Reply:
Done.